ambry-store/src/test/java/com.github.ambry.store/IndexSegmentTest.java

```diff
@@ -231,8 +292,13 @@ private void addPutEntries(List<Long> offsets, long lastEntrySize, IndexSegment
       long offset = offsets.get(i);
       long size = i == offsets.size() - 1 ? lastEntrySize : offsets.get(i + 1) - offset;
       IndexValue value =
-          new IndexValue(size, new Offset(segment.getLogSegmentName(), offset), (byte) 0, Utils.Infinite_Time);
+          IndexValueUtils.getIndexValue(size, new Offset(segment.getLogSegmentName(), offset), time.seconds(), version);
       segment.addEntry(new IndexEntry(id, value), new Offset(segment.getLogSegmentName(), offset + size));
+      if (resetKey == null) {
```

Comment:
I think `addPutEntries` should return the list of entries added so that the caller can generate the reset key if required. I would prefer if `resetKey` wasn't a class variable.